File: app/src/main/java/com/cyder/atsushi/youtubesync/json_data/SignUp.java
package com.cyder.atsushi.youtubesync.json_data;

/**
 * Created by atsushi on 2017/10/27.
 */

public class SignUp extends JsonParameter {
    public User user;

    public SignUp(String email, String name, String password) {
        user = new User(email, name, password);
    }

    private class User {
        public String email;
        public String password;
        public String name;

        public User(String email, String name, String password) {
            this.email = email;
            this.name = name;
            this.password = password;
        }
    }
}

File: app/src/main/kotlin/com/cyder/atsushi/youtubesync/api/request/SignUp.kt
package com.cyder.atsushi.youtubesync.api.request

/**
 * Created by atsushi on 2018/03/28.
 */

class SignUp(email: String, name: String, password: String) {
    private val user = User(email, name, password)

    data class User (
            val email: String,
            val name: String,
            val password: String
    )
}
